packages/java/endpoint/src/test/java/dev/hilla/crud/CrudRepositoryServiceTest.java

```diff
@@ -0,0 +1,393 @@
+package dev.hilla.crud;
+
+import dev.hilla.BrowserCallable;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.FluentQuery;
+import org.springframework.stereotype.Repository;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@SpringBootTest(classes = {
+        CrudRepositoryServiceTest.DefaultJpaRepositoryService.class,
+        CrudRepositoryServiceTest.CustomCrudRepository.class,
+        CrudRepositoryServiceTest.CustomCrudRepositoryService.class,
+        CrudRepositoryServiceTest.CustomJpaRepository.class,
+        CrudRepositoryServiceTest.CustomJpaRepositoryService.class })
+@ContextConfiguration(classes = { CrudConfiguration.class })
+@RunWith(SpringRunner.class)
+@EnableAutoConfiguration
+public class CrudRepositoryServiceTest {
+
+    @Repository
+    static class CustomCrudRepository
```

Comment:
Might be easier to run as a `@DataJpaTest` like https://github.com/vaadin/hilla/blob/main/packages/java/endpoint/src/test/java/dev/hilla/crud/FilterTest.java#L22

Reply:
It seems that `@DataJpaTest` is only initializing the Persistence Layer context, and even using it in combination with `@RunWith(SpringRunner.class)` and `@ContextConfiguration(classes = { CrudConfiguration.class })` still does not support autowiring of the service objects.